raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2653,11 +2652,11 @@ private boolean shouldFireLeaderChange(LeaderAndEpoch leaderAndEpoch) {
         }
 
         private void maybeFireLeaderChange(LeaderAndEpoch leaderAndEpoch, long epochStartOffset) {
-            // If this node is becoming the leader, then we can fire `handleClaim` as soon
+            // If this node is becoming the leader, then we can fire `handleLeaderChange` as soon
```

Comment:
Maybe useful to mention the point about the high watermark as well since it is a little subtle.

Reply:
I added a few more sentences to this comment.